frontend/packages/shared/src/components/GiteaHeader/ThreeDotsMenu/LocalChangesModal/DeleteModal/DeleteModal.tsx

```diff
@@ -19,29 +21,39 @@ export type DeleteModalProps = {
  *
  * @property {boolean}[isOpen] - If the modal is open or not
  * @property {function}[onClose] - Function to execute on close
- * @property {function}[onDelete] - Function to execute on click delete
- * @property {string}[appName] - The name of the app to delete changes on
+ * @property {string}[app] - The app
+ * @property {string}[org] - The org
  *
  * @returns {ReactNode} - The rendered component
  */
-export const DeleteModal = ({
-  isOpen,
-  onClose,
-  onDelete,
-  appName,
-}: DeleteModalProps): ReactNode => {
+export const DeleteModal = ({ isOpen, onClose, app, org }: DeleteModalProps): ReactNode => {
   const { t } = useTranslation();
 
+  const { mutate: deleteLocalChanges } = useResetRepositoryMutation(org, app);
+
   const [nameToDelete, setNameToDelete] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
```

Comment:
We should leverage the features of `@tanstack-query/react` and make use of their loading functionality to avoid dealing with manually setting state for loading and then removing the loading state. See next comment.

Reply:
Agree 😄